src/Interpreters/ConcurrentHashJoin.h

```diff
@@ -60,6 +60,13 @@ class ConcurrentHashJoin : public IJoin
     IBlocksStreamPtr
     getNonJoinedBlocks(const Block & left_sample_block, const Block & result_sample_block, UInt64 max_block_size) const override;
 
+
+    bool isCloneSupported() const override { return true; }
+    std::shared_ptr<IJoin> clone(const std::shared_ptr<TableJoin> & table_join_, const Block &, const Block & right_sample_block_) const override
+    {
+        return std::make_shared<ConcurrentHashJoin>(context, table_join_, slots, right_sample_block_, stats_collecting_params);
```

Comment:
I support it's alright, since `setTotals` and `addBlockToJoin` called only from pipeline, but clone is performed only before plan converted to pipeline. But just in case, I changed to
```cpp
    bool isCloneSupported() const override
    {
        return !getTotals() && getTotalRowCount() == 0;
    }
```